tests/tests/custom_authority_tests.cpp

```diff
@@ -1257,6 +1259,571 @@ BOOST_AUTO_TEST_CASE(custom_auths) { try {
    }
 
 
+   /**
+    * Test of authorization of one account (Alice) authorizing another key
+    * for restricted trading between between ACOIN1 and any BCOIN (BCOIN1, BCOIN2, and BCOIN3).
+    *
+    * The restricted trading authortization will be constructed with two custom authorities
+    * each of which uses one OR restriction for the BCOINs.
+    */
+   BOOST_AUTO_TEST_CASE(authorized_restricted_trading_key) {
+      try {
+         //////
+         // Initialize the blockchain
+         //////
+         generate_blocks(HARDFORK_BSIP_40_TIME);
+         generate_blocks(5);
+         db.modify(global_property_id_type()(db), [](global_property_object &gpo) {
+            gpo.parameters.extensions.value.custom_authority_options = custom_authority_options_type();
+         });
+         set_expiration(db, trx);
+
+
+         //////
+         // Initialize: Fund some accounts
+         //////
+         ACTORS((assetissuer)(alice))
+         fund(alice, asset(5000 * GRAPHENE_BLOCKCHAIN_PRECISION));
+
+
+         //////
+         // Define a key that can be authorized
+         // This can be a new key or an existing key. The existing key may even be the active key of an account.
+         //////
+         fc::ecc::private_key some_private_key = generate_private_key("some key");
+         public_key_type some_public_key = public_key_type(some_private_key.get_public_key());
+
+
+         //////
+         // Initialize: Create user-issued assets
+         //////
+         upgrade_to_lifetime_member(assetissuer);
+         create_user_issued_asset("ACOIN1", assetissuer,  UIA_ASSET_ISSUER_PERMISSION_MASK);
+         create_user_issued_asset("BCOIN1", assetissuer,  UIA_ASSET_ISSUER_PERMISSION_MASK);
+         create_user_issued_asset("BCOIN2", assetissuer,  UIA_ASSET_ISSUER_PERMISSION_MASK);
+         create_user_issued_asset("BCOIN3", assetissuer,  UIA_ASSET_ISSUER_PERMISSION_MASK);
+         create_user_issued_asset("CCOIN1", assetissuer,  UIA_ASSET_ISSUER_PERMISSION_MASK);
+         generate_blocks(1);
+         const asset_object &acoin1 = *db.get_index_type<asset_index>().indices().get<by_symbol>().find("ACOIN1");
+         const asset_object &bcoin1 = *db.get_index_type<asset_index>().indices().get<by_symbol>().find("BCOIN1");
+         const asset_object &bcoin2 = *db.get_index_type<asset_index>().indices().get<by_symbol>().find("BCOIN2");
+         const asset_object &bcoin3 = *db.get_index_type<asset_index>().indices().get<by_symbol>().find("BCOIN3");
+         const asset_object &ccoin1 = *db.get_index_type<asset_index>().indices().get<by_symbol>().find("CCOIN1");
+
+         //////
+         // Initialize: Issue UIAs
+         //////
+
+         // Lambda for issuing an asset to an account
+         auto issue_amount_to = [&](const account_id_type &issuer, const asset &amount, const account_id_type &to) {
```

Comment:
Can remove the catch-all `&` - the lamda does not depend on any outer variables.